.github/workflows/test.yaml

```diff
@@ -132,6 +132,43 @@ jobs:
       - name: Build
         run: docker buildx build -f docker/kitsune2_test_auth_hook_server/Dockerfile .
 
+  changelog-preview-comment:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout PR commits only
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          fetch-depth: ${{ github.event.pull_request.commits }}
+          sparse-checkout: ''
+          sparse-checkout-cone-mode: false
+
+      - name: Checkout all tags
+        run: git fetch --tags
+
+      - name: Install git-cliff
+        uses: taiki-e/install-action@v2
+        with:
+          tool: git-cliff@2.9.1
+
+      - name: Run git cliff
+        run: |
+          git cliff \
+            --bump \
+            --config-url=https://raw.githubusercontent.com/holochain/release-integration/refs/heads/main/pre-1.0-cliff.toml \
+            --strip=all \
+            --output=PR_CHANGELOG.md
+
+      - name: Add changelog preview as comment
+        run: |
+          gh pr comment ${{github.event.pull_request.html_url }} \
+            --body="$(printf 'The following will be added to the CHANGELOG.md\n---\n\n'; cat PR_CHANGELOG.md)" \
+            --edit-last \
+            --create-if-none
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
```suggestion
      - name: Add changelog preview as comment
        env:
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          gh pr comment ${{github.event.pull_request.html_url }} \
            --body="$(printf 'The following will be added to the CHANGELOG.md\n---\n\n'; cat PR_CHANGELOG.md)" \
            --edit-last \
            --create-if-none
```

Reply:
I find it slightly easier to digest when the env is provided before the script but just a personal preference probably :)